var/spack/repos/builtin/packages/magma/package.py

```diff
@@ -43,7 +43,7 @@ class Magma(Package):
     variant('shared',  default=True,
             description='Enables the build of shared libraries')
 
-    depends_on('cuda')
+    # depends_on('cuda') # Install CUDA manually from nvidia.developer.com
```

Comment:
This is wrong, please revert to the previous. As Todd noted you are suppose to add cuda in the packages.yaml file; this approach worked fine for me on my Mac.